src/main/java/org/jenkinsci/account/Myself.java

```diff
@@ -120,11 +120,17 @@ public HttpResponse doChangePassword(
             @QueryParameter String newPassword2
     ) throws Exception {
 
+        // verify the current password
+        try {
+            parent.connect(dn,password).close();
+        } catch (javax.naming.AuthenticationException e) {
+            throw new UserError("Wrong current password");
+        }
+
+        // verify if new password match
         if (!newPassword1.equals(newPassword2))
-            throw new Error("Password mismatch");
+            throw new UserError("New Password mismatch");
```

Comment:
Probably better to have a clearer error message, otherwise it's unclear what this is about.